### PR TITLE
Resolved TODO: platform should be a parameter

### PR DESCRIPTION
**Pull request by Valerievich**

### CODE REVIEW

File: mimesis/providers/path.py [hunk 1]
```diff
@@ -9,10 +8,10 @@
 class Path(BaseProvider):
     """Class that provides methods and property for generate paths."""
 
-    def __init__(self, *args, **kwargs):
-        # TODO: platform should be a parameter
+    def __init__(self, platform=sys.platform, *args, **kwargs):
```
Comment (sobolevn) [inline mimesis/providers/path.py]: This `platform`s should be defined as constants in https://github.com/lk-geimfari/mimesis/tree/master/mimesis/constants.

File: mimesis/providers/path.py [hunk 2]
```diff
@@ -35,7 +34,7 @@ def home(self):
         :Example:
             /home/
         """
-        if sys.platform == 'win32':
+        if self.platform == 'win32':
```
Comment (sobolevn) [inline mimesis/providers/path.py]: I would suggest not to hardcode that.

File: mimesis/providers/path.py [hunk 3]
```diff
@@ -87,5 +87,5 @@ def project_dir(self, user_gender='female'):
             /home/sherika/Development/Falcon/mercenary
         """
         project = self.random.choice(PROJECT_NAMES)
-        return os.path.join(
-            self.dev_dir(user_gender), project)
+        path = '\\' if self.platform == 'win32' else '/'
+        return (self.dev_dir(user_gender) + '{}' + project).format(path)
```
Comment (sobolevn) [inline mimesis/providers/path.py]: Why using `return ()`?

File: mimesis/providers/path.py (hunk 1 above)
Comment (sobolevn) [inline mimesis/providers/path.py]: Maybe it is a good idea to create a structure with: 
1. name
2. path separator
3. other useful things

@lk-geimfari ?

File: mimesis/providers/path.py [hunk 4]
```diff
@@ -9,10 +8,10 @@
 class Path(BaseProvider):
     """Class that provides methods and property for generate paths."""
 
-    def __init__(self, *args, **kwargs):
-        # TODO: platform should be a parameter
+    def __init__(self, platform=sys.platform, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.__p = Personal('en')
+        self.platform = platform
```
Comment (sobolevn) [inline mimesis/providers/path.py]: Could you please document all the platforms we support? What happens when user goes with `Path('does-not-exist')`?

**Comment by sobolevn** [comment]
Good work!

File: mimesis/providers/path.py (hunk 1 above)
Comment (lk-geimfari) [inline mimesis/providers/path.py]: @sobolevn Agree. @Valerievich Please, look at review.

Comment (Valerievich) [inline mimesis/providers/path.py]: @sobolevn what structure? can u please describe your vision more detail?

File: mimesis/providers/path.py (hunk 4 above)
Comment (Valerievich) [inline mimesis/providers/path.py]: ok

File: mimesis/providers/path.py (hunk 2 above)
Comment (Valerievich) [inline mimesis/providers/path.py]: Deal

File: mimesis/providers/path.py (hunk 3 above)
Comment (Valerievich) [inline mimesis/providers/path.py]: Because 'project_dir()' already had this instruction. Is it wrong?

Comment (sobolevn) [inline mimesis/providers/path.py]: Oh, now I see. It is not `return (...)`, it is some complex expression hidden there. 
It is better to reduce the complexity here.

File: mimesis/providers/path.py (hunk 1 above)
Comment (Valerievich) [inline mimesis/providers/path.py]: Do you mean constant.py should contains smth like:
```
{
    'WINDOWS': {'name': 'win32',
                'path_separator': '\\'},
    'LINUX2': {'name': 'linux2',
               'path_separator': '/'},
}
```

Comment (Valerievich) [inline mimesis/providers/path.py]: I have some idea about that structure and `Path()`. Just wait for it please.

Comment (sobolevn) [inline mimesis/providers/path.py]: And `'WINDOWS'` should be a constant as well.

File: mimesis/providers/path.py (hunk 4 above)
Comment (Valerievich) [inline mimesis/providers/path.py]: @lk-geimfari BTW, which doc.file should I update? `mimesis/docs/guide.rst` ?

Comment (sobolevn) [inline mimesis/providers/path.py]: Please, update doc-strings also.

**Comment by Valerievich** [comment]
Didn't get why `Codecov/path` were not successful

**Pull request merged**